configuration/pih/scripts/global/zl.js

```diff
@@ -0,0 +1,144 @@
+function setUpEdd(currentEncounterDate, msgWeeks) {
+  var updateEdd = function () {
+    var lastPeriodDate = htmlForm.getValueIfLegal("lastPeriodDate.value");
+    if (
+      typeof lastPeriodDate !== "undefined" &&
+      lastPeriodDate !== null &&
+      lastPeriodDate.length > 0
+    ) {
+      var today = new Date();
+      if (
+        typeof currentEncounterDate !== "undefined" &&
+        currentEncounterDate !== null &&
+        currentEncounterDate.length > 0
+      ) {
+        // calculate the gestational age at the time of the encounter
+        today = new Date(+currentEncounterDate);
+      }
+
+      var dateObj = getField("lastPeriodDate.value").datepicker("getDate");
+      var newDate = new Date(dateObj);
+      // time difference
+      var timeDiff = Math.abs(today.getTime() - newDate.getTime());
+      // weeks difference = gestational age
+      var diffWeeks = Math.ceil(timeDiff / (1000 * 3600 * 24 * 7));
```

Comment:
```suggestion
      const lastPeriodDate = new Date(getField("lastPeriodDate.value").datepicker("getDate"));
      // time difference
      const timeDiffMs = today.getTime() - lastPeriodDate.getTime();
      // weeks difference = gestational age
      const diffWeeks = Math.ceil(timeDiffMs / (1000 * 3600 * 24 * 7));
```

Use `const` when possible, and use better names. I don't know why you used `Math.abs`—if the "last period date" is in the future than we are in trouble already, but in this case a negative gestational age at least makes more sense and would make the error apparent to the user. Though I would hope the datepicker would not allow an entry in the future.